### ISSUE TITLE
unable to install using cargo

### ISSUE THREAD

**Issue by shi-yan** [tree-sitter/tree-sitter#963]
RUST_BACKTRACE=full cargo install tree-sitter-cli

```bash
    Updating crates.io index
  Installing tree-sitter-cli v0.14.5
   Compiling autocfg v1.0.1
   Compiling autocfg v0.1.7
   Compiling rand_core v0.4.2
   Compiling libc v0.2.87
   Compiling memchr v2.3.4
   Compiling cc v1.0.67
   Compiling semver-parser v0.7.0
   Compiling maybe-uninit v2.0.0
   Compiling proc-macro2 v1.0.24
   Compiling version_check v0.1.5
   Compiling unicode-xid v0.2.1
   Compiling scopeguard v0.3.3
   Compiling once_cell v1.7.2
   Compiling syn v1.0.60
   Compiling ryu v1.0.5
   Compiling serde v1.0.123
   Compiling regex-syntax v0.6.22
   Compiling bitflags v1.2.1
   Compiling hashbrown v0.9.1
   Compiling log v0.4.14
   Compiling serde_derive v1.0.123
   Compiling unicode-width v0.1.8
   Compiling serde_json v1.0.64
   Compiling cfg-if v1.0.0
   Compiling itoa v0.4.7
   Compiling byteorder v1.4.2
   Compiling strsim v0.8.0
   Compiling tree-sitter-cli v0.14.5
   Compiling vec_map v0.8.2
   Compiling lazy_static v1.4.0
   Compiling ansi_term v0.11.0
   Compiling difference v2.0.0
   Compiling smallbitvec v2.5.0
   Compiling semver v0.9.0
   Compiling rand_core v0.3.1
   Compiling rand_jitter v0.1.4
   Compiling lock_api v0.1.5
   Compiling thread_local v1.1.3
   Compiling rand_pcg v0.1.2
   Compiling rand_chacha v0.1.1
   Compiling rand v0.6.5
   Compiling num-traits v0.2.14
   Compiling num-integer v0.1.44
   Compiling indexmap v1.6.1
   Compiling num-rational v0.2.4
   Compiling textwrap v0.11.0
   Compiling nom v4.2.3
   Compiling hashbrown v0.1.8
   Compiling rustc_version v0.2.3
   Compiling rand_xorshift v0.1.1
   Compiling rand_isaac v0.1.1
error: failed to run custom build command for `tree-sitter-cli v0.14.5`

Caused by:
  process didn't exit successfully: `/tmp/cargo-installIn6GzG/release/build/tree-sitter-cli-ca72e16b35f370c5/build-script-build` (exit code: 101)
  --- stderr
  thread 'main' panicked at 'assertion failed: repo_path.pop()', /home/shiy/.cargo/registry/src/github.com-1ecc6299db9ec823/tree-sitter-cli-0.14.5/build.rs:22:13
  stack backtrace:
     0:     0x562d2f88cdc5 - backtrace::backtrace::libunwind::trace::h14d338b30b3ea0a7
                                 at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
     1:     0x562d2f88cdc5 - backtrace::backtrace::trace_unsynchronized::h73ea91d74a3fd67f
                                 at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
     2:     0x562d2f88cdc5 - std::sys_common::backtrace::_print_fmt::hd42948c952866e12
                                 at src/libstd/sys_common/backtrace.rs:78
     3:     0x562d2f88cdc5 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::ha8f928866ff7571e
                                 at src/libstd/sys_common/backtrace.rs:59
     4:     0x562d2f8aa50c - core::fmt::write::he0c1e5f7426d2718
                                 at src/libcore/fmt/mod.rs:1076
     5:     0x562d2f88a9b2 - std::io::Write::write_fmt::hf3afc6cfd57d0033
                                 at src/libstd/io/mod.rs:1537
     6:     0x562d2f88f180 - std::sys_common::backtrace::_print::hfc0110703f3696fd
                                 at src/libstd/sys_common/backtrace.rs:62
     7:     0x562d2f88f180 - std::sys_common::backtrace::print::h3f77c6990ddfaa22
                                 at src/libstd/sys_common/backtrace.rs:49
     8:     0x562d2f88f180 - std::panicking::default_hook::{{closure}}::heae49580a8d62d75
                                 at src/libstd/panicking.rs:198
     9:     0x562d2f88eecc - std::panicking::default_hook::hecc34e3f729e213c
                                 at src/libstd/panicking.rs:217
    10:     0x562d2f88f7c3 - std::panicking::rust_panic_with_hook::he82f5d0644692441
                                 at src/libstd/panicking.rs:526
    11:     0x562d2f884b5e - std::panicking::begin_panic::hb6cf2fdd818177d1
    12:     0x562d2f884100 - build_script_build::main::h49511552d13d530b
    13:     0x562d2f8848d3 - std::rt::lang_start::{{closure}}::h0cccdfb595f0ecae
    14:     0x562d2f88fb93 - std::rt::lang_start_internal::{{closure}}::h5d3ea623498f5f43
                                 at src/libstd/rt.rs:52
    15:     0x562d2f88fb93 - std::panicking::try::do_call::hac65e71be769a440
                                 at src/libstd/panicking.rs:348
    16:     0x562d2f88fb93 - std::panicking::try::hd4706e264bcf6712
                                 at src/libstd/panicking.rs:325
    17:     0x562d2f88fb93 - std::panic::catch_unwind::h948a0fb4a8b3ee82
                                 at src/libstd/panic.rs:394
    18:     0x562d2f88fb93 - std::rt::lang_start_internal::h72cc068ed2d0ac53
                                 at src/libstd/rt.rs:51
    19:     0x562d2f8848c2 - main
    20:     0x7f68b103ebf7 - __libc_start_main
    21:     0x562d2f883b4a - _start
    22:                0x0 - <unknown>
warning: build failed, waiting for other jobs to finish...
error: failed to compile `tree-sitter-cli v0.14.5`, intermediate artifacts can be found at `/tmp/cargo-installIn6GzG`

Caused by:
  build failed

```

**Comment by shi-yan** [comment]
not sure if it is related, my rust example also failed to run:

```rust
use tree_sitter::{Parser, Language};

//extern "C" { fn tree_sitter_c() -> Language; }
//extern "C" { fn tree_sitter_rust() -> Language; }
extern "C" { fn tree_sitter_javascript() -> Language; }

fn main()  {
    
    println!("debug 1");
    let mut parser = Parser::new();
    println!("debug 2");
    let language = unsafe { tree_sitter_javascript() };
    println!("debug 3 {:?}", language);
    if let Err(e) = parser.set_language(language) {
        println!("error {:?}", e);
    }
    println!("debug 4");

    let source_code = "function test(a, b) {}";
    println!("debug 5");
    let tree = parser.parse(source_code, None).unwrap();
    println!("debug 6");
    let root_node = tree.root_node();

    println!("{:?}", root_node);


}
```

I got
```
error LanguageError { version: 12 }
```

when I try to set_language.

I don't understand what it means by "version 12"

**Comment by theHamsta** [comment]
@shi-yan It means that the parser and the runtime are incompatible. Most parsers have not been updated to the new ABI version 13.

**Comment by shi-yan** [comment]
is there a parser that has been updated?

I tried the rust parser, the error is the same.

I just want to see if I can get meta data from TreeNode. For example, in addition to "identifier", I want to know which identifier that is (its name).

I don't seem to see this has been explained in the document.

**Comment by shi-yan** [comment]
Ok, I found some instructions on https://github.com/tree-sitter/tree-sitter-javascript/tree/master/bindings/rust with old parser versions. 

that worked for me.

My current conclusion is that the parser won't save meta data, such as identifiers' name, numerical literals' value etc. 

I will have to use the location references provided by the parser and read the above info from the source code.

**Comment by involuble** [comment]
The version of tree-sitter-cli on crates.io seems to be 2 years old. Would installing a newer version via git help here? (or publishing a newer version)

**Comment by silmeth** [comment]
Could you please publish new versions of `tree-sitter-cli` to crates.io? Installation using `cargo` is the first method mentioned in the [cli `README`](https://github.com/tree-sitter/tree-sitter/blob/master/cli/README.md) and the latest version in the [crates registry is `0.14.5`](https://crates.io/crates/tree-sitter-cli).

I needed a new-but-pre-0.19 version of the cli tool (to generate parsers with older ABI) and had to install it directly from github using
```
cargo install tree-sitter-cli --git ssh://git@github.com/tree-sitter/tree-sitter.git --tag v0.18.3
```

Not that it’s too problematic to install a specific tag directly from github – but if that’s required, it could be at least mentioned in the cli README.

**Comment by maxbrunsfeld** [comment]
Sorry about the delay here. I believe this is fixed, and `tree-sitter-cli` version 0.20 is on crates.io.
